packages/client/hmi-client/src/workflow/ops/calibrate-ensemble-ciemss/tera-calibrate-ensemble-node-ciemss.vue

```diff
@@ -1,225 +1,182 @@
 import {
 	getRunResultCiemss,
-	makeEnsembleCiemssCalibration,
-	querySimulationInProgress,
-	simulationPollAction
+	pollAction,
+	getCalibrateBlobURL,
+	makeEnsembleCiemssSimulation
 } from '@/services/models/simulation-service';
-import Button from 'primevue/button';
-import { RunResults } from '@/types/SimulateConfig';
 import { setupDatasetInput } from '@/services/calibrate-workflow';
-import { Poller, PollerState } from '@/api/api';
-import TeraSimulateChart from '@/workflow/tera-simulate-chart.vue';
-import TeraProgressBar from '@/workflow/tera-progress-bar.vue';
-import { getTimespan, chartActionsProxy, drilldownChartSize } from '@/workflow/util';
+import { chartActionsProxy } from '@/workflow/util';
 import { logger } from '@/utils/logger';
-import TeraOperatorPlaceholder from '@/components/operator/tera-operator-placeholder.vue';
-import {
-	CalibrateEnsembleCiemssOperation,
-	CalibrateEnsembleCiemssOperationState,
-	EnsembleCalibrateExtraCiemss
-} from './calibrate-ensemble-ciemss-operation';
+
+import { Poller, PollerState } from '@/api/api';
+import type { WorkflowNode } from '@/types/workflow';
+import { WorkflowPortStatus } from '@/types/workflow';
+import type { CsvAsset, EnsembleSimulationCiemssRequest } from '@/types/Types';
+import type { RunResults } from '@/types/SimulateConfig';
+import type { CalibrateEnsembleCiemssOperationState } from './calibrate-ensemble-ciemss-operation';
 
 const props = defineProps<{
 	node: WorkflowNode<CalibrateEnsembleCiemssOperationState>;
 }>();
-const emit = defineEmits(['append-output', 'update-state', 'open-drilldown', 'append-input-port']);
-
-const showSpinner = ref(false);
-const modelConfigIds = computed(() => getModelConfigIds);
-
-const getModelConfigIds = () => {
-	const aList: string[] = [];
-	props.node.inputs.forEach((ele) => {
-		if (ele.value && ele.type === 'modelConfigId') {
-			aList.push(ele.value[0]);
-		}
-	});
-	return aList;
-};
-const datasetId = computed(() => props.node.inputs[0].value?.[0] as string | undefined);
-const currentDatasetFileName = ref<string>();
-
-const completedRunId = ref<string>();
-const disableRunButton = computed(
-	() => !ensembleConfigs?.value[0]?.weight || !datasetId.value || !currentDatasetFileName.value
-);
-const ensembleConfigs = computed<EnsembleModelConfigs[]>(() => props.node.state.mapping);
-const extra = ref<EnsembleCalibrateExtraCiemss>(props.node.state.extra);
+const emit = defineEmits(['open-drilldown', 'update-state', 'append-output', 'append-input-port']);
 
 const runResults = ref<RunResults>({});
-const simulationIds: ComputedRef<any | undefined> = computed(
-	<any | undefined>(() => props.node.outputs[0]?.value)
-);
-const datasetColumnNames = ref<string[]>();
-const progress = ref({ status: ProgressState.Retrieving, value: 0 });
-
 const csvAsset = shallowRef<CsvAsset | undefined>(undefined);
 
-const poller = new Poller();
+const areInputsFilled = computed(() => props.node.inputs[0].value && props.node.inputs[1].value);
```

Comment:
Do these any sort of undefined/null safeties?

Reply:
I dont believe these need any safeties. I see the same check elsewhere. I get no warnings or errors when ports are not connected.
This is a boolean to determine if the user is allowed to hit the drill down button